base/system/winlogon/sas.c

```diff
@@ -1070,6 +1092,9 @@ DoGenericAction(
                     LogOffFlags |= EWX_FORCE;
                 if (!Session->Gina.Functions.WlxIsLogoffOk(Session->Gina.Context))
                     break;
+                /* Prevent LocalSystem from logging off unless it is shutting down. */
+                if (IsLocalSystem(Session->UserToken) && !WLX_SHUTTINGDOWN(wlxAction))
```

Comment:
I'm interested to know, when you test the LiveCD (and so this is the SYSTEM user being run), whether there is here a valid  `Session` pointer, and a valid (non-NULL) `Session->UserToken` value.

Reply:
If the Session->UserToken is NULL, then this is SYSTEM user.
https://git.reactos.org/?p=reactos.git;a=blob;f=base/system/winlogon/sas.c;hb=d55e33d0f3cf7ca8a1b087a677ef5726834130aa#l125
As you can see here (and there is another similar test in that file elsewhere), this is how the SYSTEM pseudo-user is currently handled in our winlogon.